modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -131,7 +131,7 @@ static CssStyleHelper createStyleHelper(final Node node) {
             }
             node.styleHelper.cacheContainer.forceSlowpath = true;
             node.styleHelper.triggerStates.addAll(triggerStates[0]);
-            node.styleHelper.firstStyleableAncestor = findFirstStyleableAncestor(node);
```

Comment:
In canReuseStyleHelper, there is this check
```
       // If the style maps are the same instance, we can re-use the current styleHelper if the cacheContainer is null.
        // Under this condition, there are no styles for this node _and_ no styles inherit.
        if (node.styleHelper.cacheContainer == null) {
            return true;
        }
```
And later in the same function is a check for (parent == null) that returns true. In both cases, firstStyleableAncestor will still point to the old first-styleable ancestor.

Reply:
Thanks. For some reason I thought all the early exits out of the method returned false. I'll push a fix.